lib/p2p/Peer.ts

```diff
@@ -451,34 +475,24 @@ class Peer extends EventEmitter {
     return packet;
   }
 
-  private handleHello = (packet: HelloPacket): void => {
+  private handleHello = (packet: packets.HelloPacket): void => {
     this.handshakeState = { ...this.handshakeState, ...packet.body };
-
-    if (this.responseMap.has(packet.type)) {
-      this.fulfillResponseEntry(packet.type, packet);
-    }
```

Comment:
I don't think this reintroduces the bug, although I need to update comments and code to make it more clear. Line 254 effectively waits for a packet with id 0, which is the hello packet, so whenever it receives that packet it will be recognized.

Reply:
I pushed a fix to an old bug I apparently left, which now makes the `Hello` bug become apparent.

HelloPacket direction is `UNILATERAL`, so when handling the packet, you don't try to fulfill the response entry. You also don't attach `reqId` to it. that's why `wait` doesn't resolve.